naga/src/front/wgsl/error.rs

```diff
@@ -820,6 +822,31 @@ impl<'a> Error<'a> {
                     "perhaps you meant to call the function in a separate statement?".into(),
                 ],
             },
+            Error::FunctionMustUseUnused(call) => ParseError {
+                message: "must use result".into(),
+                labels: vec![(call, "result unused".into())],
+                notes: vec![
+                    format!(
+                        "function '{}' is declared with `@must_use` attribute",
+                        &source[call].split_once('(').unwrap().0
+                    ),
+                    "call the function as part of an expression".into(),
+                ],
+            },
```

Comment:
I think we can make these error messages a bit better.

- Let's make the `message` more descriptive, eg "unused return value from function annotated with @must_use"
- The label with the span is good but I don't think it needs an additional description as it would just be repeating the line above, so use an empty string there.
- The notes are good, but I think we can improve the second one further by taking the suggested fix from the spec: "use a phony assignment or declare a value using the function call as the initializer."